File: 30-days-of-code/php/day-13-abstract-classes.php
<?php
abstract class Book {
    
    protected $title;
    protected  $author;
    
     function __construct($t,$a){
        $this->title=$t;
        $this->author=$a;
    }
    abstract protected function display();
}

// Begin Solution
//---------------------------------------------------------------------------------------
class MyBook extends Book {
  private $price;

  public function __construct($title, $author, $price) {
    Book::__construct($title, $author);
    $this->price = $price;
  }
  function display(){
    echo "Title: ".$this->title."Author: ".$this->author."Price: ".$this->price."\n";
  }
}
//---------------------------------------------------------------------------------------
// End Solution

$title=fgets(STDIN);
$author=fgets(STDIN);
$price=intval(fgets(STDIN));
$novel=new MyBook($title,$author,$price);
$novel->display();
?>

File: 30-days-of-code/cpp/day-13-abstract-classes.cpp
#include <string>
using namespace std;

class Book {
    protected:
        string title;
        string author;
    public:
        Book(string t,string a){
            title=t;
            author=a;
        }
        virtual void display()=0;

};

// Begin Solution
//---------------------------------------------------------------------------------------
class MyBook: public Book {
  private:
    float price;
  public:
    MyBook(string title, string author, float price) : Book(title, author){
      this->price = price;
    };
    void display(){
      cout << "Title: " << this->title << endl;
      cout << "Author: " << this->author << endl;
      cout << "Price: " << this->price << endl;
    };
    
};
//---------------------------------------------------------------------------------------
// End Solution

int main() {
    string title, author;
    int price;
    getline(cin, title);
    getline(cin, author);
    cin >> price;
    MyBook novel(title,author, price);
    novel.display();
    return 0;
}
